docs/transactions.md

```diff
@@ -1495,55 +1337,124 @@ Should be sent to trusted nodes only.
+  - time: `optional(int64)` - revocation time (number of nanoseconds elapsed since January 1, 1970 UTC). CLI uses the current time for that field.
+  - schemaVersion: `optional(uint16)` - Schema version to support backward/forward compatability(default 0)
+- In State:
+  - `pki/RevokedCertificates/value/<Certificate's Subject>/<Certificate's Subject Key ID>`
+  - `pki/RevokedNocRootCertificates/value/<Certificate's Subject>/<Certificate's Subject Key ID>`
 - CLI command:
-  - `dcld query pki revocation-points --issuer-subject-key-id=<string>`
-- REST API:
-  - GET `/dcl/pki/revocation-points/{issuerSubjectKeyID}`
+  - `dcld tx pki revoke-noc-x509-root-cert --subject=<base64 string> --subject-key-id=<hex string> --serial-number=<string> --info=<string> --time=<int64> --revoke-child=<bool> --from=<account>`
+- Validation:
+  - a NOC Root Certificate with the provided `subject` and `subject_key_id` must exist in the ledger.
 
-### GET_ALL_PKI_REVOCATION_DISTRIBUTION_POINT
+#### ADD_NOC_ICA
 
 **Status: Implemented**
 
-Gets a list of all revocation distribution points (such as RFC5280 Certificate Revocation List).
+This transaction adds a NOC ICA certificate owned by the Vendor signed by a chain of certificates which must be
+already present on the ledger.
 
-Should be sent to trusted nodes only.
+- Who can send: Vendor account
+- Validation:
+  - the provided certificate must be a non-root certificate:
+    - `Issuer` != `Subject`
+    - `Authority Key Identifier` != `Subject Key Identifier`
+  - the root certificate must be a NOC certificate and added by the same vendor
+    - `isNoc` field of the root certificate must be set to true
+    - `VID of root certificate` == `VID of account`
+  - no existing certificate with the same `<Certificate's Issuer>:<Certificate's Serial Number>` combination.
+  - if certificates with the same `<Certificate's Subject>:<Certificate's Subject Key ID>` combination already exist:
+    - the existing certificate must be NOC non-root certificate
+    - the sender's VID must match the vid field of the existing certificates.
+  - the signature and expiration date must be valid.
+- Parameters:
+  - cert: `string` - The NOC non-root Certificate, encoded in X.509v3 PEM format. Can be a PEM string or a file path.
+  - certificate-schema-version: `optional(uint16)` - Certificate's schema version to support backward/forward compatability(default 0)
+  - schemaVersion: `optional(uint16)` - Schema version to support backward/forward compatability(default 0) 
+- In State:
+  - `pki/ApprovedCertificates/value/<Subject>/<SubjectKeyID>`
+  - `pki/ApprovedCertificatesBySubject/value/<Subject>`
+  - `pki/ApprovedCertificatesBySubjectKeyID/value/<SubjectKeyID>`
+  - `pki/NocIcaCertificates/value/<VID>`
+  - `pki/ChildCertificates/value/<Certificate's Subject>/<Certificate's Subject Key ID>`
+- CLI Command:
+  - `dcld tx pki add-noc-x509-cert --certificate=<string-or-path> --from=<account>`
+
+#### REVOKE_NOC_ICA
```

Comment:
Please clarify that revoke here means soft-delete (revoke certificates can be queried)

Reply:
Done